src/NServiceBus.Extensions.Hosting/HostBuilderExtensions.cs

```diff
@@ -31,16 +34,38 @@ public static IHostBuilder UseNServiceBus(this IHostBuilder hostBuilder, Func<Ho
                 ctx.Properties.Add(HostBuilderExtensionInUse, null);
 
                 var endpointConfiguration = endpointConfigurationBuilder(ctx);
-                var startableEndpoint = EndpointWithExternallyManagedContainer.Create(endpointConfiguration, serviceCollection);
-
-                serviceCollection.AddSingleton(_ => new HostAwareMessageSession(startableEndpoint.MessageSession));
-                serviceCollection.AddSingleton<IMessageSession>(serviceProvider => serviceProvider.GetService<HostAwareMessageSession>());
-                serviceCollection.AddSingleton<IHostedService>(serviceProvider => new NServiceBusHostedService(
-                    startableEndpoint,
-                    serviceProvider,
-                    serviceProvider.GetRequiredService<ILoggerFactory>(),
-                    deferredLoggerFactory,
-                    serviceProvider.GetRequiredService<HostAwareMessageSession>()));
+
+                if (ctx.Configuration.GetValue<string>("nservicebus") == "install")
```

Comment:
at first glance, it doesn't look like the options API allows parameterless flags e.g. `--install` unless it's the very last parameter.